### PR TITLE
Add default selector.

### PR DESCRIPTION
**Pull request by wtgtybhertgeghgtwtg**
This Pull Request allows `connectModule` to be called with only the `module` and `Component` and passing the slice of the `state` belonging to `module` as the `selector`.
So
`connectModule(state => state.todo, todoModule)(todoComponent)`
can be written as
`connectModule(todoModule)(todoComponent)`
Likewise,
`connectModule(state => { todo: state.todo, counter: state.counter }, [todoModule, counterModule])(todoCounterComponent)`
can be written as
`connectModule([todoModule, counterModule])(todoCounterComponent)`

### CODE REVIEW

**Comment by mboperator** [comment]
I like the intent here. At Procore we used a pattern similar to this, with modules having a default selector. The issue is that this doesn't work with `immutable` state trees.

I'd prefer that we actually add the selector to the `createModule` object and allow the user to specify the default selector, eg:
```js
createModule({
  name: 'todos',
  selector: (state, props) => state.get('todos'),
})
```

**Comment by wtgtybhertgeghgtwtg** [comment]
Done.  It will still default to `state => state[name]` if a `selector` was not defined either in `connectModule` or `createModule`.

File: src/connectModule/index.js [hunk 1]
```diff
@@ -1,8 +1,41 @@
...
+      const selector = selectors[key];
+      props[key] = selector(state);
+    }
+    return props;
+  };
+};
+
+export const connectModule = (selector, modules) => {
+  let formattedSelector = selector;
+  let formattedModules = Array.isArray(modules) ? modules : [modules];
```
Comment (mboperator) [inline src/connectModule/index.js]: I feel like we can simplify these checks by first checking if `modules` is undefined, then checking if it's an array.

Something like
```js
if (modules) {
  formattedModules = Array.isArray(modules) ? modules : [modules];
  formattedSelector = selector;
} else {
  formattedModules = Array.isArray(selector) ? modules : [selector];
  formattedSelector = createModuleSelector(formattedModules);
}
```

Comment (wtgtybhertgeghgtwtg) [inline src/connectModule/index.js]: That could save a couple of assignments.  Done.

**Comment by mboperator** [comment]
+1 the DX benefit of this branch is huge!
At this point we're living in ~3 files: `saga`, `module`, and the `view`

**Pull request merged**